torch/lib/c10d/test/ProcessGroupNCCLErrorsTest.cpp

```diff
@@ -71,8 +72,8 @@ class ProcessGroupNCCLSimulateErrors : public c10d::ProcessGroupNCCL {
 class ProcessGroupNCCLErrorsTest : public ::testing::Test {
  protected:
   bool skipTest() {
-    // Skip test if no cuda devices found.
-    return cudaNumDevices() == 0;
+    // Skip test if no cuda devices found or NCCL version < 2.4+.
+    return cudaNumDevices() == 0 || torch::cuda::nccl::version() < 2400;
```

Comment:
This caused build failure:

```
Sep 05 20:06:57 CMakeFiles/ProcessGroupNCCLErrorsTest.dir/ProcessGroupNCCLErrorsTest.cpp.o: In function `ProcessGroupNCCLErrorsTest_testNCCLErrorsNonBlocking_Test::TestBody()':
Sep 05 20:06:57 ProcessGroupNCCLErrorsTest.cpp:(.text+0x4f1): undefined reference to `torch::cuda::nccl::version()'
```

Reply:
These files don't link against all of libtorch. It's better to use the macros here.